core/templates/dev/head/services/RteHelperService.js

```diff
@@ -181,9 +181,22 @@ oppia.factory('RteHelperService', [
           });
         });
 
-        text = elt.html();
-        text = $filter('removeExtraLines')(text);
-        return text;
+        var textElt = elt[0].childNodes;
+        for (var i = textElt.length - 1; i > 0; i--) {
+          if (textElt[i].localName === 'p') {
+            test = textElt[ i].innerHTML;
```

Comment:
This is still treating HTML content as a string. Instead the element should be analyzed with regards to child elements, and only "text nodes" in the DOM should be treated as strings.

E.g. you could look for all child elements; if all are br and all the text nodes are full of spaces only, then you can remove the paragraph DOM node.

Reply:
Also please use more clear variable names; `test` isn't very helpful. In this context it should be something like `eltHtml` (but, as per the previous comment, I recommend a different approach instead).

Reply:
@seanlip Done changes

Reply:
Doesn't look done to me -- see https://github.com/oppia/oppia/pull/4799/files